xarray/tests/test_backends.py

```diff
@@ -632,6 +632,9 @@ def test_roundtrip_boolean_dtype(self) -> None:
         with self.roundtrip(original) as actual:
             assert_identical(original, actual)
             assert actual["x"].dtype == "bool"
+            with self.roundtrip(actual) as actual2:
```

Comment:
This checks for preserving dtype during second roundtrip.

Reply:
Lets add this as a comment